lib/go/thrift/application_exception.go

```diff
@@ -77,8 +189,23 @@ func (e tApplicationException) Error() string {
 	return defaultApplicationExceptionMessage[e.type_]
 }
 
+func (e tApplicationException) Unwrap() error {
+	if e.validationError != nil {
+		return e.validationError
+	}
+	return e
```

Comment:
no you can't make `Unwrap` to return self. that will cause infinite loops. this function should simple be `return e.err`.